internal/pkg/instrumentation/bpf/net/http/server/bpf/probe.bpf.c

```diff
@@ -243,6 +252,14 @@ int uprobe_HandlerFunc_ServeHTTP(struct pt_regs *ctx)
     return 0;
 }
 
+void read_go_string(void *base, int offset, char *output, int maxLen, const char *errorMsg) {
+    void *ptr = (void *)(base + offset);
+    if (!get_go_string_from_user_ptr(ptr, output, maxLen)) {
+        bpf_printk("Failed to get %s", errorMsg);
```

Comment:
`bpfprintf` calls can be fairly expensive, we should avoid them whenever possible. 

Also, it looks like this is wrapping the `get_go_string_from_user_ptr` helper func but does some extra work by combining base and offset - is this needed or could we just use the helper direclty?

Reply:
bpf_printf call is triggered only in the case of an error, reducing the overall impact on performance.